site/_posts/2022-09-09-what-is-webauthn-why-do-you-care.md

```diff
@@ -0,0 +1,193 @@
+tags: topic-single-sign-on
+excerpt_separator: "<!--more-->"
+---
+
+WebAuthn is a new way for people to authenticate themselves to web applications. It's a widely supported standard years in the making. All major browsers work with it, which makes it easy for developers to incorporate WebAuthn into websites.
+
+WebAuthn allows users to leverage the power of biometric, device and other authentication methods via a simple, consistent, browser-native user experience.
+
+<!--more-->
+
+Am I overstating things? Is WebAuthn really all that?
+
+Read on and decide for yourself.
+
+## What is WebAuthn?
+
+WebAuthn is a way for users to authenticate to applications running in the browser. Just like a username and password, WebAuthn provides credentials, but it identifies people with a different method. WebAuthn requires a special piece of hardware or software called an "authenticator".
+
+WebAuthn removes the need for an application or user store to save off private, valuable data such as a hashed password. Instead, the application stores a public key. The corresponding private key is held securely, and in a distributed fashion, by each user's authenticator.
+
+The type of asymmetric keys that an authenticator supports, how the public key is distributed to the webapp, and how the keypair is used to authenticate a user are all specified by WebAuthn and related standards.
+
+## Why consider WebAuthn?
+
+Let's walk through a scenario to illustrate the value of WebAuthn. Say you run a website that sells clownwear (it's the next iteration of athleisure, except instead of wicking fabric, there are clown noses and big wigs). You've titled this business "Cosmo's Clown Store". You want customers and potential customers to log in to favorite items and place orders.
+
+This website is aimed at normal folks since there is not a whole lot of crossover between the geek and clownwear market, so you want to make login easy.
+
+However, it is also an ecommerce company. If someone's credentials are stolen, they could be on the hook for the illicit credit card transactions, and your brand will suffer. You need to balance authentication security with an easy user experience. 
+
+In doing your research, you come across a couple of options. You wonder "Why would I choose to implement WebAuthn? Aren't the current solutions for credentials just fine?"
+
+### What's wrong with current options
+
+Let's briefly look at common solutions for securing customer accounts:
+
+* Passwords
+* Password managers
+* Multi-factor authentication (MFA)
+
+Passwords are a well known technology. They've been around since the 1960s and have the virtue of being well understood by most users. Users remember a username and password pair, and provide those to the application during the authentication process to prove who they are. However, passwords have a built-in tension. For each and every user, they must choose a password that is memorable, but difficult to guess. That's a difficult balance.
```

Comment:
```suggestion
Passwords are a well-known technology. They've been around since the 1960s and have the virtue of being well understood by most users. Users remember a username and password pair, and provide those to the application during the authentication process to prove who they are. However, passwords have a built-in tension. For each and every user, they must choose a password that is memorable, but difficult to guess. That's a difficult balance.
```